qa/L0_backend_python/python_test.py

```diff
@@ -122,50 +127,56 @@ def _optional_input_infer(self, model_name, has_input0, has_input1):
-        # Total shared memory available is 1GiB.
-        total_byte_size = 1024 * 1024 * 1024
-        shape = [total_byte_size]
-        with self.assertRaises(InferenceServerException) as ex:
-            self._infer_help(model_name, shape, dtype)
-        self.assertIn(
-            "Failed to increase the shared memory pool size", str(ex.exception)
-        )
+        # NOTE: Windows tests are not running in a docker container. Consequently, we
+        # do not specify a --shm-size to use a basis to grow. Therefore, this test does
+        # not apply for Windows.
+        if not _test_windows:
+            # 2 MiBs
+            total_byte_size = 2 * 1024 * 1024
+            shape = [total_byte_size]
+            model_name = "identity_uint8_nobatch"
+            dtype = np.uint8
+            with self._shm_leak_detector.Probe() as shm_probe:
+                self._infer_help(model_name, shape, dtype)
+
+            # 1 GiB payload leads to error in the main Python backend process.
+            # Total shared memory available is 1GiB.
+            total_byte_size = 1024 * 1024 * 1024
+            shape = [total_byte_size]
+            with self.assertRaises(InferenceServerException) as ex:
+                self._infer_help(model_name, shape, dtype)
+            self.assertIn(
+                "Failed to increase the shared memory pool size", str(ex.exception)
+            )
 
-        # 512 MiBs payload leads to error in the Python stub process.
-        total_byte_size = 512 * 1024 * 1024
-        shape = [total_byte_size]
-        with self.assertRaises(InferenceServerException) as ex:
-            self._infer_help(model_name, shape, dtype)
-        self.assertIn(
-            "Failed to increase the shared memory pool size", str(ex.exception)
-        )
+            # 512 MiBs payload leads to error in the Python stub process.
+            total_byte_size = 512 * 1024 * 1024
+            shape = [total_byte_size]
+            with self.assertRaises(InferenceServerException) as ex:
+                self._infer_help(model_name, shape, dtype)
+            self.assertIn(
+                "Failed to increase the shared memory pool size", str(ex.exception)
+            )
 
-        # 2 MiBs
-        # Send a small paylaod to make sure it is still working properly
-        total_byte_size = 2 * 1024 * 1024
-        shape = [total_byte_size]
-        with self._shm_leak_detector.Probe() as shm_probe:
-            self._infer_help(model_name, shape, dtype)
+            # 2 MiBs
+            # Send a small paylaod to make sure it is still working properly
+            total_byte_size = 2 * 1024 * 1024
+            shape = [total_byte_size]
+            with self._shm_leak_detector.Probe() as shm_probe:
+                self._infer_help(model_name, shape, dtype)
 
     # GPU tensors are not supported on jetson
     # CUDA Shared memory is not supported on jetson
-    if not TEST_JETSON:
+    if not _test_jetson and not _test_windows:
```

Comment:
This could be in a follow up ticket. We should use a [decorator](https://docs.python.org/3.10/library/unittest.html#unittest.skipIf) for skipping tests on different platforms.

Reply:
Created DLIS-6129 to track the effort to address this comment and your previous ones.